VK/VK/Controllers/FriendsTableViewController.swift

```diff
@@ -105,34 +106,32 @@ final class FriendsTableViewController: UITableViewController {
                 self.configureListFriends()
                 self.tableView.reloadData()
             case let .failure(error):
-                print(error)
+                self.showAlert(title: nil, message: error.localizedDescription, actionTitle: nil, handler: nil)
             }
         }
     }
 
-    private func fetchRealmFriends() {
-        do {
-            print(realm?.configuration.fileURL ?? "")
-            guard let friends = realm?.objects(Friend.self) else { return }
-            addToken(friends)
-            userFriends = friends
-            fetchFriends()
-        } catch {
-            print(error.localizedDescription)
+    private func loadData() {
+        realmService.loadData { [weak self] friends in
+            guard let self = self else { return }
+            // guard friends: Results<Friend> else { return }
+            self.userFriends = friends
+            self.addNotificationToken(friends)
+            self.fetchFriends()
         }
     }
 
-    private func addToken(_ result: Results<Friend>) {
+    private func addNotificationToken(_ result: Results<Friend>) {
         guard let userFriends = userFriends else { return }
-        notificationToken = userFriends.observe { [weak self] changes in
+        notificationToken = userFriends.observe { changes in
```

Comment:
```suggestion
        notificationToken = userFriends.observe { [weak self] changes in
```
я же написал не удаляй)